src/routers/user.js

```diff
@@ -24,4 +26,10 @@ router
     userController.deleteUser
   );
 
+// POST and PUT routes for /api/users/rating/:id
+router
+  .route('/rating/:userid')
+  .post(userAuthenticationMiddleware, userController.rateUser)
+  .put(userAuthenticationMiddleware, userController.updateUserRating);
```

Comment:
A better route structure would be `/:userid/rating`.
Then the complete route will become `/api/users/:userid/rating` which can be read right to left as "rating of user id of users collection".